code/modules/projectiles/gun_system.dm

```diff
@@ -413,11 +413,12 @@
 		O.emp_act(severity)
 
 /obj/item/weapon/gun/equipped(mob/user, slot)
-	unwield(user)
 	if(ishandslot(slot))
 		set_gun_user(user)
-		return ..()
-	set_gun_user(null)
+		mouse_opacity = MOUSE_OPACITY_OPAQUE
+	else
+		set_gun_user(null)
+	unwield(user)
```

Comment:
Why'd you move this? Check procs before moving them around, this effects the user, so if you're unsetting gun user before calling this you're going to break a bunch of stuff.